extensions/ql-vscode/src/model-editor/model-evaluator.ts

```diff
@@ -0,0 +1,73 @@
+import type { ModelingStore } from "./modeling-store";
+import type { ModelingEvents } from "./modeling-events";
+import type { DatabaseItem } from "../databases/local-databases";
+import type { ModelEvaluationRun } from "./model-evaluation-run";
+import { DisposableObject } from "../common/disposable-object";
+import { sleep } from "../common/time";
+import type { ModelEvaluationRunState } from "./shared/model-evaluation-run-state";
+
+export class ModelEvaluator extends DisposableObject {
+  public constructor(
+    private readonly modelingStore: ModelingStore,
+    private readonly modelingEvents: ModelingEvents,
+    private readonly dbItem: DatabaseItem,
+    private readonly updateView: (
+      run: ModelEvaluationRunState,
+    ) => Promise<void>,
+  ) {
+    super();
+
+    this.registerToModelingEvents();
+  }
+
+  public async startEvaluation() {
+    // Update store with evaluation run status
+    const evaluationRun: ModelEvaluationRun = {
+      isPreparing: true,
+      variantAnalysisId: undefined,
+    };
+    this.modelingStore.updateModelEvaluationRun(this.dbItem, evaluationRun);
+
+    // For now, just wait 5 seconds and then update the store with the succeeded status.
+    // In the future, this will be replaced with the actual evaluation process.
```

Comment:
We can probably remove the comment about "succeeded" status, since we now just use the `isPreparing` boolean